### PR TITLE
Use Culture Specified by User in case it differs with that of OS

### LINKED ISSUE #821: How to change display language to english?
I once used Win10&VS2017 in chinese, so I got display language in CMD was chinese, it looked like:

> Microsoft (R) 测试执行命令行工具版本 15.0.0.0 版权所有 (C) Microsoft
> Corporation。保留所有权利。
> 
> 正在启动测试执行，请稍候...
> ConsoleApp.Sqlite.Test.PrimeServiceIsPrimeShould.ReturnFalseGivenValueOf1
> 个失败 错误消息: Unable to get default constructor for class
> ConsoleApp.Sqlite.Test.PrimeServiceIsPrimeShould.
> 
> 总测试: 1。已通过: 0。失败: 1。已跳过: 0。 测试运行失败。 测试执行时间: 0.5729 秒

Then I changed my Win10's & VS2017's language to english, everything was fine (display in english) except the VSTest tools, it still displayed in chinese.

So my question is, how chould I change the language of VSTest tools to english?

Cache related? or language package of .Net Framework?

### PR DESCRIPTION
**Pull request by mayankbansal018**
## Description
Set Current Culture to be the one propagated by dotnet via `DOTNET_CLI_UI_LANGUAGE`

Test:
set DOTNET_CLI_UI_LANGUAGE=fr
run vstest.console  /?

You should see help displayed in french

## Related issue
Fixes #821.

### CODE REVIEW

File: src/datacollector/DataCollectorMain.cs [hunk 1]
```diff
@@ -140,6 +141,23 @@ private void WaitForDebuggerIfEnabled()
...
+            var userCultureSpecified = Environment.GetEnvironmentVariable("DOTNET_CLI_UI_LANGUAGE");
+            if (!string.IsNullOrWhiteSpace(userCultureSpecified))
+            {
+                try
+                {
+                    CultureInfo info = new CultureInfo(userCultureSpecified);
+                    CultureInfo.DefaultThreadCurrentUICulture = info;
+                }
+                catch (Exception)
+                {
```
Comment (smadala) [inline src/datacollector/DataCollectorMain.cs]: Do EqtTrace here.  `ConsoleOutput` not useful in datacollector process.

File: src/testhost.x86/Program.cs [hunk 2]
```diff
@@ -103,5 +104,22 @@ private static void WaitForDebuggerIfEnabled()
                 Debugger.Break();
             }
         }
+
+        private static void SetCultureSpecifiedByUser()
+        {
+            var userCultureSpecified = Environment.GetEnvironmentVariable("DOTNET_CLI_UI_LANGUAGE");
```
Comment (smadala) [inline src/testhost.x86/Program.cs]: DRY.

File: src/vstest.console/Program.cs [hunk 3]
```diff
@@ -36,7 +37,26 @@ public static int Main(string[] args)
                 System.Diagnostics.Debugger.Break();
             }
 
+            SetCultureSpecifiedByUser();
+
             return new Executor(ConsoleOutput.Instance).Execute(args);
         }
+
+        private static void SetCultureSpecifiedByUser()
```
Comment (smadala) [inline src/vstest.console/Program.cs]: DRY.

File: src/datacollector/DataCollectorMain.cs [hunk 4]
```diff
@@ -140,6 +141,23 @@ private void WaitForDebuggerIfEnabled()
...
 
+        private static void SetCultureSpecifiedByUser()
+        {
+            var userCultureSpecified = Environment.GetEnvironmentVariable("DOTNET_CLI_UI_LANGUAGE");
+            if (!string.IsNullOrWhiteSpace(userCultureSpecified))
+            {
+                try
+                {
+                    CultureInfo info = new CultureInfo(userCultureSpecified);
+                    CultureInfo.DefaultThreadCurrentUICulture = info;
```
Comment (smadala) [inline src/datacollector/DataCollectorMain.cs]: Add example testing to PR description.

**Review by smadala — CHANGES_REQUESTED** [review]

File: src/vstest.console/Program.cs (hunk 3 above)
Comment (mayankbansal018) [reply src/vstest.console/Program.cs]: Specify where it should be common for all these 3 different application

File: src/testhost.x86/AppDomainEngineInvoker.cs [hunk 5]
```diff
@@ -167,6 +170,25 @@ private static string GetConfigFile(string testSource, string testSourceFolder)
...
 
+        private static void SetAppDomainCulture(string[] args)
+        {
+            var userCultureSpecified = Environment.GetEnvironmentVariable(CoreUtilities.Constants.DotNetUserSpecifiedCulture);
+            if (!string.IsNullOrWhiteSpace(userCultureSpecified))
+            {
+                try
+                {
+                    CultureInfo info = new CultureInfo(userCultureSpecified);
+                    CultureInfo.DefaultThreadCurrentCulture = info;
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/testhost.x86/AppDomainEngineInvoker.cs]: done

**Review by smadala — APPROVED** [review]
Please add required tests.

**Comment by mayankbansal018** [comment]
added tests for app domain culture

**Pull request merged**